sdformat_mjcf/sdformat_mjcf/converters/material.py

```diff
@@ -0,0 +1,99 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module to convert SDFormat MAterials to MJCF"""
+
+from ignition.math import Vector3d
+
+from sdformat import Pbr, PbrWorkflow, Material  # noqa: F401
+
+import sdformat_mjcf.sdf_utils as su
+
+import os
+
+TEXTURE_NUMBER = 0
+
+
+def add_material(body, material):
+    """
+    Converts an SDFormat material to an MJCF material and add it to the given
+    body.
+
+    :param mjcf.Element body: The MJCF body to which the geom is added.
+    :param sdf.Material material: The SDF material to convert
+    :return: The newly created MJCF material.
+    :rtype: mjcf.Element
+    """
+    pbr = material.pbr_material()
+    sp_color = material.specular()
+    em_color = material.emissive()
+    specular = (sp_color.r() + sp_color.g() + sp_color.b()) / 3.0
+    emissive = (em_color.r() + em_color.g() + em_color.b()) / 3.0
+    asset = body.root.asset
+    if pbr is not None:
+        workflow = pbr.workflow(PbrWorkflow.PbrWorkflowType.METAL)
+        if workflow is not None:
+            if workflow.albedo_map():
+                extension_tokens = os.path.basename(
+                    workflow.albedo_map()).split(".")
+                if (len(extension_tokens) == 0):
+                    raise RuntimeError("Unable to find the extension {}"
+                                       .format(workflow.albedo_map()))
+                file_without_extension = os.path.splitext(
+                    os.path.basename(workflow.albedo_map()))[0]
+
+                texture_loaded = asset.find('texture',
+                                            file_without_extension)
+                if texture_loaded is None:
+                    texture = asset.add('texture',
+                                        name=file_without_extension,
+                                        type='2d',
+                                        file=workflow.albedo_map(),
+                                        gridsize='1 1')
+                    return asset.add("material",
+                                     name="material_" + file_without_extension,
+                                     texture=texture,
+                                     texrepeat="1 1",
+                                     texuniform=True,
+                                     specular=specular,
+                                     emission=emissive)
+                else:
+                    return asset.find('material',
+                                      'material_' + file_without_extension)
+    else:
+        global TEXTURE_NUMBER
+        diffuse = material.diffuse()
+        ambient = material.ambient()
+        texture = asset.add('texture',
+                            name="texture_" + str(TEXTURE_NUMBER),
```

Comment:
Setting only rgba

![Simulate : shapes_072](https://user-images.githubusercontent.com/1933907/167726392-0c13e811-6e92-48c7-9e3f-93304f594258.png)


Settting rgb1 and rgb2

![Simulate : shapes_073](https://user-images.githubusercontent.com/1933907/167726432-fe3a52ad-51f8-472a-aa97-86d6dc7675c6.png)

Reply:
I think that's happening because it's using the `checker` texture. Instead, for solid colors, we don't need any texture. Just setting the rgba on the material should work.

Reply:
We can probably use the logic from https://github.com/gazebosim/sdformat/pull/526#discussion_r623937715 to comvine ambient and diffuse to get the `rgba`.

Reply:
https://github.com/gazebosim/gz-mujoco/pull/18/commits/16c652980e7484bdaabcf816c334f80103952777